### ISSUE TITLE
`manual_div_ceil` does not trigger for alternate form

### ISSUE THREAD

**Issue by ronnodas** [rust-lang/rust-clippy#16940]
### Summary

The lint triggers for expressions of the form `(a + b - 1) / b` but not of the form `(a - 1) / b + 1`. For unsigned integers, these are equivalent ignoring overflow/underflow.

### Lint Name

manual_div_ceil

### Reproducer

I tried this code ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=947a084643102c623404db89c59ea852)):

```rust
fn div_ceil(a: u32, b: u32) -> u32 {
    (a - 1) / b + 1
}
```

I expected to see this happen:
The `manual_div_ceil` lint should trigger, suggesting `a.div_ceil(b)`.

Instead, this happened:
Nothing.

**Comment by Gri-ffin** [comment]
the two expressions arent equivalent in all cases since rust division truncates towards zero for signed ints, take the case of a = -1, and b = 4. we get 0 for both the first expression and `div_ceil` but 1 for the second.

**Comment by ronnodas** [comment]
@Gri-ffin thanks, edited to specify `a` and `b` unsigned.
